fuel-bft/src/error.rs

```diff
@@ -0,0 +1,45 @@
+use core::convert::Infallible;
+use core::fmt;
+
+/// Consensus error variants
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
```

Comment:
why not use thiserror?

Reply:
https://github.com/FuelLabs/fuel-bft/issues/12